### PR TITLE
Updated Builder test with agent

### PR DESCRIPTION
**Pull request by Loghijiaha**
## Checklist

- [x] I have referenced the Jira issue related to my changes in one or more commit messages
- [x] Unit tests pass locally with my changes
- [ ] I have added documentation as necessary
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [ ] I have interactively tested my changes

## Types of changes

- [x] Infrastructure change (non-breaking change which updates dependencies or improves infrastructure)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by Loghijiaha** [comment]
The last commit (Pipeline test) successfully passes tests in my local machine. @kinow If you have a little time to run mvn clean test, check with this PR.

File: src/test/java/io/jenkins/plugins/ml/IPythonBuilderTest.java [hunk 1]
```diff
@@ -68,18 +72,25 @@ public void createMocks() throws Exception {
...
-        String name = "%text 37";
-        IPythonBuilder builder = new IPythonBuilder("32+5",null,"text");
-        project.getBuildersList().add(builder);
-        FreeStyleBuild build = jenkins.buildAndAssertSuccess(project);
-        jenkins.assertLogContains( name, build);
-        */
+        p.setDefinition(
+                new CpsFlowDefinition(
+                        "node{\n"
+                                + "def testImage = docker.image('loghijiaha/ml-agent')\n"
```
Comment (Loghijiaha) [inline src/test/java/io/jenkins/plugins/ml/IPythonBuilderTest.java:84]: @kinow How can we setup python dependencies in CI server by using Jenkins file? I have used our ml-agent image to start the container. The script is running inside the docker. Your error is different than the CI errors. I am a little confused.

Comment (kinow) [reply src/test/java/io/jenkins/plugins/ml/IPythonBuilderTest.java:84]: I think all the dependencies wikm have to be installed in the container. Nothing in the box running the CI build.

Comment (kinow) [reply src/test/java/io/jenkins/plugins/ml/IPythonBuilderTest.java:84]: Or maybe there is something in the Jenkins Wiki about that. I've used Jenkins in-house only. On GH always Travis or GH actions. So not sure how to troubleshoot errors like this.

**Comment by oleg-nenashev** [comment]
So `docker.image('loghijiaha/ml-agent')` makes the nested steps to run in your container provisioned as agent. 

> How can we setup python dependencies in CI server by using Jenkins file?

https://github.com/jenkins-infra/documentation/blob/master/ci.adoc provides some information. Generally the dependencies are restricted, and the builds are using users with restricted permissions. Running a container agent allows to work it around, but in the case of Docker Pipeline Plugin, again, a restricted user is used b default. Likely this is why it cannot find `pip` in https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fmachine-learning-plugin/detail/PR-47/4/tests

You may grant the user with id 1000 admin permissions in your instance to work it around. You can also force the container to run as root. E.g. see https://stackoverflow.com/questions/42630894/jenkins-docker-how-to-control-docker-user-when-using-image-inside-command

**Pull request merged**